extension/jaegerremotesampling/internal/remote_strategy_store.go

```diff
@@ -0,0 +1,62 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"context"
+	"fmt"
+
+	grpcstore "github.com/jaegertracing/jaeger/cmd/agent/app/configmanager/grpc"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type grpcRemoteStrategyStore struct {
+	conn            *grpc.ClientConn
+	delegate        *grpcstore.SamplingManager
+	headerAdditions map[string]configopaque.String
+}
+
+// NewRemoteStrategyStore returns a StrategyStore that delegates to the configured Jaeger gRPC endpoint, making
+// extension-configured enhancements (header additions only for now) to the gRPC context of every outbound gRPC call.
+// Note: it would be nice to expand the configuration surface to include an optional TTL-based caching behavior
+// for service-specific outbound GetSamplingStrategy calls.
+func NewRemoteStrategyStore(
+	ctx context.Context,
+	host component.Host,
+	telemetry component.TelemetrySettings,
+	grpcClientSettings *configgrpc.GRPCClientSettings,
+) (strategystore.StrategyStore, func() error, error) {
+	conn, err := grpcClientSettings.ToClientConn(ctx, host, telemetry)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error while connecting to the remote sampling source: %w", err)
+	}
+	return &grpcRemoteStrategyStore{
+			conn:            conn,
+			delegate:        grpcstore.NewConfigManager(conn),
+			headerAdditions: grpcClientSettings.Headers,
+		},
+		func() error {
+			return conn.Close()
+		},
+		nil
+}
+
+func (g grpcRemoteStrategyStore) GetSamplingStrategy(ctx context.Context, serviceName string) (*sampling.SamplingStrategyResponse, error) {
+	return g.delegate.GetSamplingStrategy(g.enhanceContext(ctx), serviceName)
+}
+
+// This function is used to add the extension configuration defined HTTP headers to a given outbound gRPC call's context.
+func (g grpcRemoteStrategyStore) enhanceContext(ctx context.Context) context.Context {
```

Comment:
To be very reductive, this function is the core of the PR: adding HTTP headers (as configured) to outgoing gRPC calls (as context metadata).

Reply:
```suggestion
func (g *grpcRemoteStrategyStore) enhanceContext(ctx context.Context) context.Context {
```

Maybe we should use a ptr receiver here to be consistent.